src/Util/BatchRequest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Http\Client\Util;
+
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Http\Client\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Implements sending multiple request for client not supporting parallels requests
+ *
+ * Internally, use the sendRequest method
+ *
+ * Should be used with Http\Client\HttpPsrClient
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+trait BatchRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function sendRequest(RequestInterface $request, array $options = []);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequests(array $requests, array $options = [])
+    {
+        $batchResult    = new BatchResult();
+        $batchException = new BatchException();
+        $batchException->setResult($batchResult);
```

Comment:
We can create the exception on demand but i think it will make code harder to read, as we must add some conditions / loop system as there is no way to pass an array of request -> exceptions to the BatchException in its current implementation